File: src/main/java/com/bindstone/kuljetus/converter/CategorieConverter.java
package com.bindstone.kuljetus.converter;

import com.bindstone.kuljetus.domain.enumeration.Categorie;
import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.ReadingConverter;

@ReadingConverter
public class CategorieConverter implements Converter<String, Categorie> {

    @Override
    public Categorie convert(String s) {
        return Categorie.byCode(s);
    }
}

File: src/main/java/com/bindstone/kuljetus/converter/CategorieConverter.kt
package com.bindstone.kuljetus.converter

import com.bindstone.kuljetus.domain.enumeration.Categorie
import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.ReadingConverter

@ReadingConverter
class CategorieConverter : Converter<String, Categorie> {
    override fun convert(s: String): Categorie {
        return Categorie.byCode(s)
    }
}
